src/main/java/com/labsynch/labseer/service/BulkLoadServiceImpl.java

```diff
@@ -2391,31 +2393,30 @@ private Parent updateParentAliases(Parent parent, Parent foundParent, String che
 		// Note: This is not the most efficient / concise code to do this process
 		// however due to readability and the assumingly small number of aliases 
 		// parents will have this is fine 
-
-		Set<String> parentAliasStrings = new HashSet<>();
+		HashMap<String, ParentAlias> newAliases = new HashMap<String, ParentAlias>();
 		for(ParentAlias alias : parent.getParentAliases()){
 			if (! alias.isIgnored()) {
-				parentAliasStrings.add(alias.getAliasName());
+				newAliases.put(alias.getAliasName(), alias);
 			}
 		}
-		Set<String> foundParentAliasStrings = new HashSet<>(); 
+		HashMap<String, ParentAlias> existingParentAliases = new HashMap<String, ParentAlias>();
 		for(ParentAlias alias : foundParent.getParentAliases()){
 			if (! alias.isIgnored()) {
-				foundParentAliasStrings.add(alias.getAliasName());
+				existingParentAliases.put(alias.getAliasName(), alias);
 			}
 		}
-
-		Set<ParentAlias> unionParentAliases = parent.getParentAliases();
```

Comment:
We make a set of the new *ParentAlias*es. These all lack the `id` property, i.e. they're all new Java objects that have not been persisted to the DB.